invenio_records_marc21/services/components/metadata.py

```diff
@@ -21,6 +22,9 @@ class MetadataComponent(BaseMetadataComponent):
     def create(self, identity, data=None, record=None, errors=None, **kwargs):
         """Inject parsed metadata to the record."""
         record.metadata = data.get("metadata", {})
+        if errors:
+            messages = list({str(e) for e in errors})
```

Comment:
minor: better [str(e) for e in errors] instead of converting dictionary into list